### PR TITLE
ADEN-9245 Revert to using two callbacks for accept or reject

### PR DESCRIPTION
**Pull request by mklucsarits**
In order to accommodate existing logic on platforms using the GDPR dialog, we need to revert to using two separate callbacks from the dialog -- onAcceptTracking and onRejectTracking. These will be called when non-IAB vendor tracking is either accepted or rejected. The user-allowed IAB vendors and purposes will be passed as parameters to either callback as arrays.

Additionally I've fixed a race condition when initializing the CMP by utilizing an existing Promise.

### CODE REVIEW

File: src/TrackingOptIn.js [hunk 1]
```diff
@@ -114,6 +118,7 @@ class TrackingOptIn {
             preventScrollOn: this.options.preventScrollOn,
         };
 
+console.log('HERE 1', this.hasUserConsented());
```
Comment (nandy-andy) [inline src/TrackingOptIn.js]: debugging?

File: src/TrackingOptIn-test.js [hunk 2]
```diff
@@ -115,7 +118,9 @@ describe('TrackingOptIn', () => {
         trackingOptIn.render();
 
         assert.isNotOk(modalIsShown());
-        assert.isOk(onRejectTracking.called);
+        assert.isOk(consentManagementProvider.install.called);
+        // TODO assert function called after promise
+        // assert.isOk(onRejectTracking.called);
```
Comment (JoshuaRogan) [inline src/TrackingOptIn-test.js]: would a timeout work?

Comment (JoshuaRogan) [reply src/TrackingOptIn-test.js]: outside of that, I don't see any way besides properly setting up promise testing https://www.chaijs.com/plugins/chai-as-promised/

Comment (mklucsarits) [reply src/TrackingOptIn-test.js]: I was overthinking. It's as easy as calling a stubbed function that resolves the promise and asserting the other function call with `then`.

**Pull request merged**